service/history/callback_queue_active_task_executor.go

```
@@ -0,0 +1,268 @@
+			return nil, serviceerror.NewInternal(fmt.Sprintf("failed to construct Nexus completion: %v", err))
+		}
+		return completion, nil
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+		failure := commonnexus.APIFailureToNexusFailure(ce.GetWorkflowExecutionFailedEventAttributes().GetFailure())
+		return &nexus.OperationCompletionUnsuccessful{
+			State:   nexus.OperationStateFailed,
+			Failure: failure,
+		}, nil
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED:
+		return &nexus.OperationCompletionUnsuccessful{
+			State:   nexus.OperationStateCanceled,
+			Failure: &nexus.Failure{Message: "operation canceled"},
+		}, nil
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
+		return &nexus.OperationCompletionUnsuccessful{
+			State:   nexus.OperationStateFailed,
+			Failure: &nexus.Failure{Message: "operation terminated"},
+		}, nil
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+		return &nexus.OperationCompletionUnsuccessful{
+			State:   nexus.OperationStateFailed,
+			Failure: &nexus.Failure{Message: "operation exceeded internal timeout"},
+		}, nil
+	}
+	return nil, queues.NewUnprocessableTaskError(fmt.Sprintf("invalid workflow execution status: %v", ce.GetEventType()))
+}
+
+func (t *callbackQueueActiveTaskExecutor) processNexusCallbackTask(ctx context.Context, task *tasks.CallbackTask, url string, completion nexus.OperationCompletion) (retErr error) {
+	request, err := nexus.NewCompletionHTTPRequest(ctx, url, completion)
+	if err != nil {
+		return serviceerror.NewInternal(fmt.Sprintf("failed to construct Nexus request: %v", err))
+	}
+
+	caller := t.callerProvider(task.NamespaceID, task.DestinationAddress)
+	response, callErr := caller(request)
+	var callback *persistencespb.CallbackInfo
+	wfCtx, release, ms, err := t.getValidatedMutableStateForTask(
+		ctx, task, func(ms workflow.MutableState) error {
+			var err error
+			callback, err = t.validateCallbackTask(ms, task)
+			return err
+		},
+	)
+	if err != nil {
+		return nil
```

Is this swallowing the validation error? Could you add a quick comment why that error should not be returned?

---

Oops it was supposed to return the error, thanks!